src/App.tsx

```diff
@@ -15,35 +16,136 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => {
+        return a.length - b.length;
+      });
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
+
 export const App: React.FC = () => {
+  const [sortOptions, setSortOptions] = useState<State>({
+    isReversed: false,
+    sortType: SortType.NONE,
+  });
+  const sortAlphabetically = () => {
+    setSortOptions(prevState => ({
+      ...prevState,
+      sortType: SortType.ALPHABET,
+    }));
+  };
+
+  const sortByLength = () => {
+    setSortOptions(prevState => ({
+      ...prevState,
+      sortType: SortType.LENGTH,
+    }));
+  };
+
+  const reset = () => {
+    setSortOptions(() => ({
+      isReversed: false,
+      sortType: SortType.NONE,
+    }));
```

Comment:
here you don't need arrow function. We use it when require new state to be based on previous one 
```suggestion
    setSortOptions({
      isReversed: false,
      sortType: SortType.NONE,
    });
```